### PR TITLE
Improve level set reinitialization

### LINKED ISSUE #14849: Improve the reinitialization solution of level set
## Reason

The solution quality of the reinitialization case needs to be improved. 

## Design

Another simpler algorithm that seems to be working better than the current one will be implemented. 

## Impact

The simpler algorithm will be set as default and some tests need regloden.

### PR DESCRIPTION
**Pull request by jiangwen84**
close #14849

### CODE REVIEW

**Comment by jiangwen84** [comment]
@aeslaughter Could you review it?

File: modules/level_set/test/tests/reinitialization/tests [hunk 1]
```diff
@@ -1,13 +1,23 @@
...
+    requirement = "The level set module shall be capable of solving the level set equation with original reinitialization."
+    issues = '#14849 #'
+  [../]
+  [./full_modified_reinitialization]
+    # Runs two steps of a full modified reinitialization problem just to make sure it continues to work
+    type = Exodiff
+    input = master.i
+    exodiff = master_modified.e
+    cli_args = 'Executioner/num_steps=2 MultiApps/reinit/input_files=reinit_modified.i Outputs/file_base=master_modified'
+    method = 'opt' # complete reinitilization solve, debug is slow and not needed
```
Comment (permcody) [inline modules/level_set/test/tests/reinitialization/tests]: FYI - I prefer to see '!dbg' when possible. The reason is that some developers like to work in "devel" or "oprof" modes and this test would still run.

Comment (jiangwen84) [reply modules/level_set/test/tests/reinitialization/tests]: done!

File: modules/level_set/test/tests/reinitialization/tests [hunk 2]
```diff
@@ -1,13 +1,23 @@
...
+  [../]
+  [./full_modified_reinitialization]
+    # Runs two steps of a full modified reinitialization problem just to make sure it continues to work
+    type = Exodiff
+    input = master.i
+    exodiff = master_modified.e
+    cli_args = 'Executioner/num_steps=2 MultiApps/reinit/input_files=reinit_modified.i Outputs/file_base=master_modified'
+    method = '!dbg' # complete reinitilization solve, debug is slow and not needed
+    requirement = "The level set module shall be capable of solving the level set equation with modified reinitialization."
+    issues = '#8465 #'
```
Comment (aeslaughter) [inline modules/level_set/test/tests/reinitialization/tests]:
```suggestion
    issues = '#14849'
  [../]
  [./full_modified_reinitialization]
    # Runs two steps of a full modified reinitialization problem just to make sure it continues to work
    type = Exodiff
    input = master.i
    exodiff = master_modified.e
    cli_args = 'Executioner/num_steps=2 MultiApps/reinit/input_files=reinit_modified.i Outputs/file_base=master_modified'
    method = '!dbg' # complete reinitilization solve, debug is slow and not needed
    requirement = "The level set module shall be capable of solving the level set equation with modified reinitialization."
    issues = '#8465'
```

**Review by aeslaughter — CHANGES_REQUESTED** [review]

**Comment by aeslaughter** [comment]
Thanks, just one little typo.

**Review by aeslaughter — APPROVED** [review]

**Comment by jiangwen84** [comment]
@aeslaughter I want to include two gifs in the markdown file. What would be the best way to upload them to large_media? sent to you or submit a PR there? 

vortex_modified
vortex_original

**Pull request merged**